.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,92 @@
+name: Deploy backend functions and frontend app
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  path-changes:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            convert_ids:
+              - "functions/convert_ids/convert_ids_deploy/**"
+            ml:
+              - "functions/ml/ml_deploy/**"
+    outputs:
+      convert_ids: ${{ steps.changes.outputs.convert_uds }}
+      ml: ${{ steps.changes.outputs.ml }}
+
+  deploy-convert-ids:
+    needs: path-changes
+    if: ${{ needs.path-changes.outputs.convert_ids == 'true' }}
+    uses: ./.github/workflows/helper-deploy-func.yaml
+    with:
+      func-name: "gpz-convert-ids"
+      func-src-dir: "functions/convert_ids/convert_ids_deploy"
+      func-entrypoint: "convert_ids"
+      func-memory-mb: 1024
+      func-data-gcs-url: "gs://geneplexus-func-data/convert-ids/convert-ids_data.tar.gz"
+    secrets: inherit
+
+  deploy-ml:
+    needs: path-changes
+    if: ${{ needs.path-changes.outputs.ml == 'true' }}
+    uses: ./.github/workflows/deploy-func.yaml
+    with:
+      func-name: "gpz-ml"
+      func-src-dir: "functions/ml/ml_deploy"
+      func-entrypoint: "ml"
+      func-memory-mb: 8192
+      func-data-gcs-url: "gs://geneplexus-func-data/ml/ml_data.tar.gz"
+    secrets: inherit
+
+  deploy-frontend:
+    needs:
+      - deploy-convert-ids
+      - deploy-ml
```

Comment:
Neat; I was concerned that the `deploy-frontend` step might get skipped if either of these didn't run, but it seems that the `if` statement takes priority and explicitly checks for whether they're either successful or skipped. Nice solution!